service/src/lib.rs

```diff
@@ -174,11 +174,9 @@ pub fn new_partial<RuntimeApi, Executor, Extrinsic>(config: &mut Configuration,
 		Executor: NativeExecutionDispatch + 'static,
 		Extrinsic: RuntimeExtrinsic,
 {
-	if !test {
-		// If we're using prometheus, use a registry with a prefix of `polkadot`.
-		if let Some(PrometheusConfig { registry, .. }) = config.prometheus_config.as_mut() {
-			*registry = Registry::new_custom(Some("polkadot".into()), None)?;
-		}
+	// If we're using prometheus, use a registry with a prefix of `polkadot`.
+	if let Some(PrometheusConfig { registry, .. }) = config.prometheus_config.as_mut() {
+		*registry = Registry::new_custom(Some("polkadot".into()), None)?;
```

Comment:
I also removed the test flag. It was useful before when this was a macro. But now it doesn't matter because it's using the imports in the current crate.